lib/resty/upstream/healthcheck.lua

```diff
@@ -597,6 +605,16 @@ function _M.spawn_checker(opts)
         return nil, "no upstream specified"
     end
 
+    local hook_down = opts.hook_down
```

Comment:
iirc, LuaJIT discourages such caching practices (temporarily unwrapping a table's value in a local variable), warning to not try to second-guess the compiler: table lookups are already optimized and we should only cache long-lived table references.

I could be wrong ofc.

Source: http://wiki.luajit.org/Numerical-Computing-Performance-Guide